static/js/containers/App.js

```diff
@@ -88,17 +89,14 @@ class App extends React.Component {
       location: { pathname },
       dispatch,
     } = this.props;
-    let [ complete, info ] = validateProfileComplete(profile);
+    let [ complete, url, errors] = validateProfileComplete(profile);
     if (
       userProfile.getStatus === FETCH_SUCCESS &&
       profile.agreed_to_terms_of_service &&
       !PROFILE_REGEX.test(pathname) &&
       !complete
     ) {
-      const { url, title, text } = info;
-      dispatch(updateDialogText(text));
-      dispatch(updateDialogTitle(title));
-      dispatch(setDialogVisibility(true));
+      dispatch(updateProfileValidation(errors));
```

Comment:
the change you need to make is right here.

before we can dispatch the `updateProfileValidation` action we need to have already dispatched the `startProfileEdit` action - otherwise there is no in-progress 'edit' object (which we keep separate in the store from the object pulled from the API) to store the validation errors in.

So with what you have here they aren't being cleared, they're simply being dropped on the floor.